src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from copy import deepcopy
+from argparse import Namespace
+import numpy as np
+import torch
+from tensorboardX import SummaryWriter
+
+from nni.compression.torch.compressor import Pruner
+from .channel_pruning_env import ChannelPruningEnv
+from .lib.agent import DDPG
+from .lib.utils import get_output_folder
+
+torch.backends.cudnn.deterministic = True
+
+class AMCPruner(Pruner):
+    """
+    A pytorch implementation of AMC: AutoML for Model Compression and Acceleration on Mobile Devices.
+    (https://arxiv.org/pdf/1802.03494.pdf)
+
+    Parameters:
+        model: nn.Module
+            The model to be pruned.
+        config_list: list
+            Configuration list to configure layer pruning.
+            Supported keys:
+                op_types: operation type to be pruned
+                op_names: operation name to be pruned
+        evaluator: function
+            function to evaluate the pruned model.
+            The prototype of the function:
+                >>> def val_func(val_loader, model):
+                >>>     ...
+                >>>     return acc
+        val_loader: torch.utils.data.DataLoader
+            Data loader of validation dataset.
+        suffix: str
+            suffix to help you remember what experiment you ran
+        export: boolean
+            False: search best pruned model and export after search.
+            True: export a searched model, before exporting model, you must run this pruner
+        export_path: str
+            path for exporting models
+
+        # parameters for pruning environment
+        model_type: str
+            model type to prune, currently 'mobilenet' and 'mobilenetv2' are supported.
+        sparsity: float
+            prune ratio
```

Comment:
this one is not in config_list but as input argument?

Reply:
changed `sparsity` to `flops_ratio` which means preserve flops ratio.